### ISSUE TITLE
Is the `installsAfter` property set on a Feature ignored during scenario testing?

### ISSUE THREAD

**Issue by eitsupi** [devcontainers/cli#347]
It seems that the `installsAfter` property set in devcontainer-feature.json is not respected during the scenario tests and the test fails, is this a known limitation?

https://github.com/eitsupi/rocker-devcontainer-features/blob/174de74153714a8f57e576ca175a8fd230aaded3/src/r-packages/devcontainer-feature.json#L29-L34
https://github.com/eitsupi/rocker-devcontainer-features/blob/174de74153714a8f57e576ca175a8fd230aaded3/test/r-packages/scenarios.json#L31-L39
https://github.com/rocker-org/devcontainer-features/actions/runs/3801508456/jobs/6466055972

**Comment by eitsupi** [comment]
Maybe due to #345?

**Comment by samruddhikhandale** [comment]
I tested one scenario where the `installsAfter` is respected. https://github.com/devcontainers/features/blob/main/src/java/devcontainer-feature.json#LL80

![image](https://user-images.githubusercontent.com/24955913/209987131-091e2120-4391-43b5-8352-6b61dac45d32.png)

In this case, the Features are referenced only with `id` as they exist in the same `src` Folder.

@eitsupi  In your case, can you update the scenario in [here](https://github.com/eitsupi/rocker-devcontainer-features/blob/174de74153714a8f57e576ca175a8fd230aaded3/test/r-packages/scenarios.json#L34) & replace `ghcr.io/rocker-org/devcontainer-features/r-apt:latest` with `r-apt` ? That should unblock you for now. 

The current scenario which fails for you will be fixed once https://github.com/devcontainers/cli/pull/335 is released.

**Comment by eitsupi** [comment]
Thanks for the quick reply.
I have a question, if I specify the name `r-apt`, which is different from the name specified in `installsAfter` property (`ghcr.io/rocker-org/devcontainer-features/r-apt:latest`), will it still change the order of installation?

**Comment by samruddhikhandale** [comment]
> I have a question, if I specify the name r-apt, which is different from the name specified in installsAfter property (ghcr.io/rocker-org/devcontainer-features/r-apt:latest), will it still change the order of installation?

I believe the `installsAfter` property is respected as both of the Features are in same Folder. Hence, `ghcr.io/rocker-org/devcontainer-features/r-apt:latest` should correspond with `r-apt`

@joshspicer Wondering if you might have other thoughts.

**Comment by joshspicer** [comment]
Only published Features (with a fully qualified `ghcr`/etc ID can be used in the `installsAfter` property.  To execute the tests, each Feature being tested is added to an autogenerated `devcontainer.json` as a "local Feature" and referenced by relative path.  

Other Features in the scenario that have something like `ghcr.io/../r-apt` in their `installsAfter` property would not have that respected when running the tests.

This could definitely be an area for improvement if folks find it useful, this behavior is an artifact of how the test command is implemented.

**Comment by samruddhikhandale** [comment]
With the changes in https://github.com/devcontainers/cli/pull/335

It is now possible to do something like 👇 

<img width="700" alt="image" src="https://user-images.githubusercontent.com/24955913/209998138-985f900f-f319-4564-b4c3-2ef8e15d2f2f.png">

The `installsAfter` is respected. Sample changes 👉  https://github.com/devcontainers/features/compare/samruddhikhandale/tmp?expand=1

<img width="832" alt="image" src="https://user-images.githubusercontent.com/24955913/209998570-5747e1e9-056b-4982-829e-455cf8314e77.png">

**Comment by BusHero** [comment]
It seems like `installsAfter` gets ignored in all scenarios, not just testing.
Here is a [repo](https://github.com/BusHero/devcontainer-bug-installafter) with devcontainer with [dotnet](https://github.com/devcontainers/features/tree/main/src/dotnet) feature.
[test.sh](https://github.com/BusHero/devcontainer-bug-installafter/blob/main/test.sh) shows that [common-utils](https://github.com/devcontainers/features/tree/main/src/common-utils) feature does not get installed in the devcontainer.

**Comment by eitsupi** [comment]
@BusHero You may be misunderstanding.
`installsAfter` is for changing the installation order of Features' listed in the `features` property, not for installing Features that are not in `features`.

**Comment by BusHero** [comment]
Ok, now I see. So if I specify in the `devcontainer.json`
```json
"features": {
	"ghcr.io/devcontainers/features/dotnet": {},
	"ghcr.io/devcontainers/features/common-utils": {}
}
```
`installsAfter` will guarantee that `common-utils` will be installed first

**Comment by samruddhikhandale** [comment]
Yep, that's correct.

**Comment by samruddhikhandale** [comment]
> The current scenario which fails for you will be fixed once https://github.com/devcontainers/cli/pull/335 is released.

Released the fix with v0.27.0 🚀 
Feel free to re-open this issue if the bug resurfaces, thanks!
